pkg/daemon/ceph/client/filesystem.go

```diff
@@ -381,3 +381,63 @@ func GetMDSDump(context *clusterd.Context, clusterInfo *ClusterInfo) (*MDSDump,
 	}
 	return &dump, nil
 }
+
+// SubvolumeGroup is a representation of a Ceph filesystem subvolume group.
+type SubvolumeGroup struct {
+	Name string `json:"name"`
+}
+
+// SubvolumeGroupList is the representation Ceph returns when listing Ceph filesystem subvolume groups.
+type SubvolumeGroupList []SubvolumeGroup
+
+// ListSubvolumeGroups lists all subvolume groups in a given filesystem by name.
+// Times out after 5 seconds.
+var ListSubvolumeGroups = listSubvolumeGroups
+
+// with above, allow this to be overridden for unit testing
+func listSubvolumeGroups(context *clusterd.Context, clusterInfo *ClusterInfo, fsName string) (SubvolumeGroupList, error) {
+	svgs := SubvolumeGroupList{}
+
+	args := []string{"fs", "subvolumegroup", "ls", fsName}
+	cmd := NewCephCommand(context, clusterInfo, args)
+	buf, err := cmd.RunWithTimeout(5 * time.Second)
+	if err != nil {
+		return svgs, errors.Wrapf(err, "failed to list subvolumegroups in filesystem %q", fsName)
+	}
+
+	if err := json.Unmarshal(buf, &svgs); err != nil {
+		return svgs, errors.Wrapf(err, "failed to unmarshal subvolumegroup list for filesystem %q", fsName)
+	}
+
+	return svgs, nil
+}
+
+// Subvolume is a representation of a Ceph filesystem subvolume.
+type Subvolume struct {
+	Name string `json:"name"`
+}
+
+// SubvolumeList is the representation Ceph returns when listing Ceph filesystem subvolumes.
+type SubvolumeList []Subvolume
+
+// ListSubvolumesInGroup lists all subvolumes present in the given filesystem's subvolume group by
+// name. Times out after 5 seconds.
+var ListSubvolumesInGroup = listSubvolumesInGroup
+
+// with above, allow this to be overridden for unit testing
+func listSubvolumesInGroup(context *clusterd.Context, clusterInfo *ClusterInfo, fsName, groupName string) (SubvolumeList, error) {
+	svs := SubvolumeList{}
+
+	args := []string{"fs", "subvolume", "ls", fsName, groupName}
```

Comment:
Perhaps the perf question above pertains more to subvolumes than the subvolume groups